listeners.go

```diff
@@ -128,16 +129,27 @@ func PublicInterfaceAddrs() ([]string, error) {
 	})
 }
 
+// Wrapper to keep track of allocated servers per host
+type krazyServer struct {
+	srv  *http.Server
+	port string
+}
+
 var (
-	listeners   = make(map[string]*http.Server)
+	listeners   = make(map[string][]*krazyServer)
 	listenersMu sync.Mutex
 )
 
-func updateListeners(port string) error {
+// tlsEnabled: Indicate, if public or private addresses should be used
```

Comment:
I wonder whether private/public address selection should really be tied to TLS being enabled.

Of course, when TLS is not enabled, HTTP should not be available on public addresses.
But, just because TLS is enabled, doesn’t mean I necessarily want to serve HTTP on public addresses.
Think of the unencrypted WiFi use-case: updating should still not necessarily be possible via the WAN.

Maybe we need another knob? It might be time to introduce a JSON config file with parameters for the gokrazy web interface. To cover the port-related TODO above, maybe we allow specifying the listen address.

Reply:
I'd prefer the latter from a developer's perspective as well as from user's/administrator's perspective. For example: I live in a dormitory with a big routed network with private legacy IPs. Just allowing "private" addresses doesn't make a difference for me, as the service would be still reachable for about 40k people (including the university).

Reply:
> I'd prefer the latter

Sorry, when you say “the latter”, what do you mean?



> I live in a dormitory with a big routed network with private legacy IPs. Just allowing "private" addresses doesn't make a difference for me, as the service would be still reachable for about 40k people (including the university).

Right, but it also doesn’t hurt, right? I.e., you don’t necessarily need to use a public address, you can just keep using a private address (even though in your particular network private isn’t as private as elsewhere)?

Reply:
> > I'd prefer the latter
> 
> Sorry, when you say “the latter”, what do you mean?

Allowing the user to specify the listen address(es). In my opinion, this would make the behavior of the server more "defined".
Should "listen private only" still be the fallback, if not defined otherwise or if the address is not configured on any interface?